### PR TITLE
More archboard feedback

### PR DESCRIPTION
**Pull request by m-nash**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Review by ShivangiReja — APPROVED** [review]

**Review by christothes — APPROVED** [review]
core LGTM

File: sdk/core/Azure.Core.TestFramework/src/Instrumentation/ManagementInterceptor.cs [hunk 1]
```diff
@@ -13,18 +13,20 @@ namespace Azure.Core.TestFramework
...
             _testBase = testBase;
+            _testMode = testBase is RecordedTestBase recordedTestBase ? recordedTestBase.Mode : RecordedTestMode.Playback;
         }
 
         public void Intercept(IInvocation invocation)
         {
             bool modifiedAskToWait = false;
 
-            if (IsLro(invocation.Method.ReturnType))
+            if (_testMode == RecordedTestMode.Playback && IsLro(invocation.Method.ReturnType))
```
Comment (christothes) [inline sdk/core/Azure.Core.TestFramework/src/Instrumentation/ManagementInterceptor.cs:29]: should we create a simple test to validate this with a simple LRO?

File: sdk/core/Azure.Core/tests/ManagementRecordedTestBaseTests.cs [hunk 2]
```diff
@@ -203,5 +203,27 @@ public void ValidateForwardClientCallDefault()
...
+            Mode = mode;
+            ManagementTestClient testClient = InstrumentClient(new ManagementTestClient());
+            TestResource testResource = testClient.GetTestResource();
+            Stopwatch sw = Stopwatch.StartNew();
+            testResource = (await testResource.GetLroAsync(WaitUntil.Completed)).Value;
+            sw.Stop();
+            Assert.AreEqual("TestResourceProxy", testResource.GetType().Name);
+            if (mode == RecordedTestMode.Playback)
+            {
+                Assert.LessOrEqual(sw.ElapsedMilliseconds, 1000);
```
Comment (christothes) [inline sdk/core/Azure.Core/tests/ManagementRecordedTestBaseTests.cs:221]: I think the Assert is ambiguous if both are xOrEqual to 1000

```suggestion
                Assert.Less(sw.ElapsedMilliseconds, 1000);
```

File: sdk/core/Azure.Core.TestFramework/src/Instrumentation/OperationInterceptor.cs [hunk 3]
```diff
@@ -55,10 +55,17 @@ internal static object InvokeWaitForCompletion(object target, Type targetType, C
         {
             // get the concrete instance of OperationPoller.ValueTask<Response<T>> WaitForCompletionAsync<T>(Operation<T>, TimeSpan?, CancellationToken)
             var poller = InjectZeroPoller();
-            var genericMethod = poller.GetType().GetMethods().Where(m => m.Name == PollerWaitForCompletionAsyncName).FirstOrDefault(m => m.GetParameters().Length == 3);
+            var genericMethod = poller.GetType().GetMethods().Where(m => m.Name == PollerWaitForCompletionAsyncName).FirstOrDefault(m => m.GetParameters().Length == 6);
```
Comment (archerzz) [inline sdk/core/Azure.Core.TestFramework/src/Instrumentation/OperationInterceptor.cs:58]: Here we invoke `OperationPoller.WaitForCompletionAsync(Operation operation, TimeSpan? timeSpan?, CancellationToken cancellationToken)` which will expand to the 6 parameter version. `UdpateStatusAsync`, `HasCompleted`, `Value`, `GetRawResponse` all come from the `operation` which we mock.
So here it should be OK to invoke this method? I roll back the codes and run the test locally. It seems fine.

File: sdk/core/Azure.Core/tests/ManagementRecordedTestBaseTests.cs [hunk 4]
```diff
@@ -203,5 +203,37 @@ public void ValidateForwardClientCallDefault()
...
+                testResource = (await testResource.GetLroAsync(WaitUntil.Completed)).Value;
+                sw.Stop();
+                Assert.AreEqual("TestResourceProxy", testResource.GetType().Name);
+                if (mode == RecordedTestMode.Playback)
+                {
+                    Assert.LessOrEqual(sw.ElapsedMilliseconds, 1000);
+                }
+                else
+                {
+                    Assert.GreaterOrEqual(sw.ElapsedMilliseconds, 1000);
```
Comment (archerzz) [inline sdk/core/Azure.Core/tests/ManagementRecordedTestBaseTests.cs:230]: The test failed on CI with the following error message. I think it's a noise, so we could add some tolerance.
> Error Message:
     Expected: greater than or equal to 1000
  But was:  999

```suggestion
                    Assert.That(sw.ElapsedMilliseconds, Is.GreaterThanOrEqualTo(1000).Within(50));
```

**Pull request merged**